### PR TITLE
DIP-19

### PR DESCRIPTION
**Pull request by costgallo**

### CODE REVIEW

**Comment by costgallo** [comment]
Created this DIP for hosting DevCon videos archive on Swarm via Etherna in addition to the current offerings

File: DIPs/DIP-17.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+## Motivation & Rationale
+__Implementing this DIP would make DevCon’s website hosting decentralized and would make DevCon’s digital efforts aligned with its principles and values.__
+
+## Implementation
+- Has any part of this proposal been implemented at other events? If so, please describe how it went.
+Not yet. We are planning to reach out to more events that happen before to implement this solution. This DIP will be updated accordingly. 
+However, the solution has been implemented on other situations like blogposts: 
+http://fairdatasociety.bzz.link/
+https://fairdataprotocol.bzz.link/
```
Comment (ligi) [inline DIPs/DIP-17.md:34]: the first one 404s for me - the second one works

Comment (ligi) [reply DIPs/DIP-17.md:34]: image

Comment (ligi) [reply DIPs/DIP-17.md:34]: maybe we should not use the 404ing one as (the first) example then

File: DIPs/DIP-17.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+- Do you require feedback or data from attendees post-event?
+No
+
+## Operational Requirements & Ownership
+__Please answer questions below:__
+1. What actions are required to implement the proposal at Devcon?
+
+We need to set up a Bee light or full node with all the necessary data pinned on them. Also we need to set up a gateway-proxy that is able to:
+Provide a DNSLink (optional) so that it can serve regular DNS domains (e.g. devcon.ethereum.org) but still having the data in a format that is compatible with Swarm.
```
Comment (ligi) [inline DIPs/DIP-17.md:46]: we do things for devcon on devcon.org not ethereum.org

**Review by ligi — CHANGES_REQUESTED** [review]
Thanks for the proposal - I really like the sentiment - but think in the beginning we need to find an hybrid web2/web3 approach for the website - similar as we do for the videos in the archive

File: DIPs/DIP-19.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+Costanza Gallo will act as the bridge between the DevCon team, the Swarm Association and Etherna.
+
+
+## 3. What other projects could this proposal be integrated with? (Bonus points for collaboration across teams :))
+This is a joint project between Swarm and Etherna. 
+
+# Links & Additional Information
+
+__Any additional information__
+
```
Comment (ligi) [inline DIPs/DIP-19.md:100]:
```suggestion
```

**Review by ligi — CHANGES_REQUESTED** [review]
Thanks! Great proposal - just small nits
would also be nice to get some more details on the interface to the DEVCon videos / metadata required - could be the start of what I had in mind with a "DEVCon/Event API" a while ago. But we need to know what the interface will be.

**Comment by ligi** [comment]
can you remove DIP-17 from here? was merged in the other PR

here a sample json: https://ipfs-cluster.ethdevops.io/ipfs/QmQr1GhBMqp6b7zH8ouM9Mkp3nvLDkoMhygGxYrfhB7g7J/Game%20Day%20Remix_berlagezaal_1.json
for the metadata for videos - this could be reused for easier integration - but a good standard/API for this is badly missing.

**Comment by ligi** [comment]
can you remove DIP-17 from here? was merged in the other PR

here a sample json: https://ipfs-cluster.ethdevops.io/ipfs/QmQr1GhBMqp6b7zH8ouM9Mkp3nvLDkoMhygGxYrfhB7g7J/Game%20Day%20Remix_berlagezaal_1.json
for the metadata for videos - this could be reused for easier integration - but a good standard/API for this is badly missing.

**Comment by agazso** [comment]
> can you remove DIP-17 from here? was merged in the other PR

Removed the DIP-17, it is now possible to resolve your comments on that file.

**Review by ligi — APPROVED** [review]

File: DIPs/DIP-19.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
+---
+DIP: DIP-19
```
Comment (ligi) [inline DIPs/DIP-19.md:2]:
```suggestion
DIP: 19
```

File: DIPs/DIP-19.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
+---
+DIP: 19
+Title: Hosting DevCon Archive Videos on Swarm via Etherna
+Status: Draft
+Themes: Virtual Experience, Website, Community Involvement
+Tags: Software, Event Operations, Communications
+Authors: Costanza Gallo costanza@ethswarm.org
+Resources Required: Operation support
+Discussion: https://forum.devcon.org/t/host-devcon-videos-on-swarm-etherna/585/6
```
Comment (ligi) [inline DIPs/DIP-19.md:9]:
```suggestion
Discussion: https://forum.devcon.org/t/host-devcon-videos-on-swarm-etherna/585/6
Created: 2023-08-26
```

File: DIPs/DIP-19.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+This is needed for be able to offer data even if Swarm garbage collect them. Best to have, even if Swarm pinning could be managed at any time after the upload.
+
+#### Update Postage Stamps management (August-September) - Nice to have
+
+Currently every user has a single postage stamp for every upload. We want to update this, and create a new postage stamp for each upload group. In this way a single video, it's encoded versions, it's thumbnail ecc, would be uploaded with the same postage stamp, and every group with a different one. This offer a better granularity when postage stamps need to be renewed.
+
+#### Test and devs for remove arbitrary limit of 500MB upload (end of Septemper) - Nice to have 
+
+Currently an arbitrary limit of 500MB for uploaded file size has been implemented.
+More tests have to been done, and probably a gateway component (interceptor) will need to be rewrited with a different technology (migrate from Nginx to a reverse proxy implemented with .Net). Anyway in this phase videos are still manually encoded, and compression can be adjusted accordly.
```
Comment (ligi) [inline DIPs/DIP-19.md:65]:
```suggestion
More tests have to been done, and probably a gateway component (interceptor) will need to be rewritten with a different technology (migrate from Nginx to a reverse proxy implemented with .Net). Anyway in this phase videos are still manually encoded, and compression can be adjusted accordingly.
```

**Review by ligi — CHANGES_REQUESTED** [review]
Small typos

**Review by ligi — APPROVED** [review]

**Comment by tmm360** [comment]
Hello everyone, as Etherna we would like to provide video hosting based on Swarm also for Devcon VII, with improved tools and infrastructure!

**Pull request merged**